docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,132 @@
+
+Consider a typical `Online Shopping` site. First the UI requirements are gathered. Customers would be provided a home page, a product list, a shopping cart and an order history page.
+
+Next, the API requirements are assessed. If the capability to choose products, add to cart and submit an order are exposed as secure APIs, they can be used by an in-house web application as well as a third party mobile app developer. 
+
+Once the API requirements are assessed,
+
+- Front end designers would want to try the `Online Shopping` UI experience.
+  > Loopback provides options to configure simple methods as remote API endpoints.
+
+- API developers would want to assess API specifications and security options.
+  > Loopback provides options to decorate remote methods with security options.
+
+- Back end developers would want to create Entity-Relationship models from backend datasource
+  > Loopback provides extensive features to connect to backend and generate object models from datasource schemas.
+
+The example application [loopback4-example-shopping](https://github.com/strongloop/loopback4-example-shopping)
+is a nice place for developers to start looking on how to use LoopBack 4.
+
+In the following sections we will see how LoopBack can be used for the online shopping example.
+
+# Usecases for Online Shopping
+
+For our `Online Shopping` site how the customer might interact with a web or mobile application is the same.
+
+- `Login usecases`
+  - The customer would register as a user.
+  - The customer logs in as a user.
+  - After the user logs in, the mobile/web app creates a shopping cart for the user.
+
+- `Shopping usecases`
+  - User starts adding and removing items to the shopping cart.
+  - If the user logs out, the shopping cart is saved and used for the next user session.
+
+- `Order usecases`
+  - User creates an order with items in the shopping cart.
+  - The mobile/web app submits the order and resets the shopping cart.
+  - User takes a look at all the previous orders.
+
+We have the following API requirements for our `Online Shopping` site from the above usecases:
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+| user creates a shopping cart  | PUT /shoppingCarts/{userId}          | create or reset shopping cart  |
+| user adds items to cart       | POST /shoppingCarts/{userId}/items | updates shopping cart  |
+| user removes an item from the cart | DEL /shoppingCarts/{userId}/items/{itemId} | removes an item from cart |
+| user submits the order | POST /users/{userId}/orders | returns order object  |
+| user's list of all orders | GET /users/{userId}/orders | returns all orders linked with user |
+
```

Comment:
Can we add a section to examine backend realities:

- User records and orders are stored in databases
- Production recommendation is an internal service that exposes REST or gRPC APIs
- Shopping cart is backed by a Redis registry
- Security is enforced with User database and ...

Reply:
There is an internal product recommendation service that has REST/gRPC APIs for shopping app to use.